ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/MicrometerMetricsPrometheus.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.Application
+import io.ktor.application.ApplicationCall
+import io.ktor.application.call
+import io.ktor.application.feature
+import io.ktor.http.ContentType
+import io.ktor.http.charset
+import io.ktor.http.content.OutgoingContent
+import io.ktor.metrics.micrometer.AbstractMicrometerMetrics
+import io.ktor.response.ApplicationSendPipeline
+import io.ktor.response.respond
+import io.ktor.routing.get
+import io.ktor.routing.routing
+import io.ktor.util.cio.bufferedWriter
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig
+import io.micrometer.prometheus.PrometheusConfig
+import io.micrometer.prometheus.PrometheusMeterRegistry
+import io.prometheus.client.Collector
+import io.prometheus.client.exporter.common.TextFormat
+import kotlinx.coroutines.io.ByteWriteChannel
+import java.util.*
+
+class MicrometerMetricsPrometheus(config: Configuration) :
```

Comment:
After removing unnecessary functionality, the redundant inheritance is even more obvious. I believe prometheus should be installed into the original micrometer metrics functionality. There is no need to inherit it. We need a separate Prometheus config and a routing function to install an endpoint. `meterRegistry` extension property could return it from the original feature as well.